### ISSUE TITLE
Shift of margincitations when citing in a sidenote

### ISSUE THREAD

**Issue by smetzler** [fmarotta/kaobook#67]
Hi, I very much like the new feature introduced in #46 and #50 . In fact, I had my own (somewhat more primitive) definition of a \sidetextcite before, and just now started to switch to yours. It seems worth it, since I noticed that the placement of the margincitations now significantly improved: There is no need manually shift them around anymore, so that they don't overlap. 

The issue I have is regarding this placement: I'm having a page where the margin is filled with one sidenote at the top and lots of margincitations thereafter. The alignment is all fine, i.e. citations occur right next to their mentioning in the text, whenever possible, sometimes a bit further down, when the space was already taken. Now, I want to additionally cite a paper within the sidenote. 

Expected outcome: 
The additional citation shows under the sidenote. The positions of the following citations remain unaffected as much as possible. 

Observed outcome: 
The additional citation takes the place of the formerly first citation on the page, causing a shift of every citation to the next position. The original positioning however is kept, such that now citation x is displayed next to the mentioning of citation x+1. The last citation of the page is even pushed to the following page.

How do I prevent this?
Let me know if you need more information, such as an illustration or a minimal example.

Cheers,
Saskia

**Comment by bradyneal** [comment]
I'd guess that this largely has to do with just LaTeX floating handling things. The current way I know of to get the behavior you want is to manually add offsets to all of them. Basically, any time a margin is super full, you have two options (in my experience):
1. Let floating deal with all of them however it decides
2. Give up your morning, and manually place each of them ;)

I'd guess that the solution you want will be difficult to automate, but we'll see what kaomaster @fmarotta says.

**Comment by fmarotta** [comment]
I largely agree with @bradyneal, but above all I'm surprised that you didn't get a "Float lost" error: are you maybe using an older version of the template?

For the next ~10 days I will take a break from LaTeX (sorry about that), but, for my return, it would certainly be helpful for me to have a minimal working example to play with, just to understand whether there is a bug in the template, or if a workaround can be found.

Right now, the only workaround I could think of is not to use `\sidecite` inside of the `\sidenote`, but rather, to print the citation marker in the sidenote, and then create a separate marginnote with the citation "body".

**Comment by smetzler** [comment]
Thanks for the prompt responses. I'll try to do an example the next days where I make sure to work on the current version of the template. For my current project, style files are up to date, but I wouldn't bet for any left over code in a hidden corner. 

My experience for very full margins is exactly what @bradyneal mentions. Here, however the margin is not filled up. I attach an illustration of the issue.

Citation [6] is not displayed here, and everything is fine:
![without](https://user-images.githubusercontent.com/69352579/91493841-2e802d00-e8b8-11ea-90c7-e26b7f4dafac.png)
The result of \sidetextcite within the sidenote:
![sidetextcite](https://user-images.githubusercontent.com/69352579/91493933-55d6fa00-e8b8-11ea-8a4b-3a8edbfe58c6.png)
The expected result (I used the former version of \sidecite with a manual shift to achieve the workaround):
![expected](https://user-images.githubusercontent.com/69352579/91494331-f3322e00-e8b8-11ea-9fde-93dc25ed172e.png)


Sorry for the double posting, btw. That's what you get when you try to use github on an outdated browser...

**Comment by smetzler** [comment]
I get the "Floats lost" error, as suspected by @fmarotta, when trying to create a minimal example on the basis of the report-template.

**Comment by fmarotta** [comment]
Sorry for the late reply. It turns out that the fix for #68 might also have affected this problem: I managed to avoid the shift issue by passing an offset to `\sidetextcite` when calling it inside the sidenote, i.e.
```
\sidenote{For similar structures, \sidetextcite[1cm]{Borgatti1999} coined the terms...}
```
The Borgatti citation now displays just below the sidenote (note that you may need to adjust the offset argument to match the length of your sidenote), and the other citations in the margin are unaffected.

One more thing: if you specify the offset for a marginnnote (or a margin citation), the automatic mechanism that prevents overlaps fails, so you may have to specify the offset for other marginnotes as well. (This may be the topic of another issue, though.)

Hope this helps, let me know in any case!
